code/modules/modular_computers/file_system/programs/generic/reports.dm

```diff
@@ -115,16 +115,16 @@
 	if(href_list["save"])
 		if(!selected_report)
 			return 1
-		if(!selected_report.verify_access(get_access(user)))
+		if(!(selected_report.get_file_perms(get_access(user), user) & OS_READ_ACCESS))
```

Comment:
Should this be `WRITE`? The href is 'save'.

Reply:
Technically no since this currently always saves a copy. However looking at the UI it seems like saving as a copy should just be an option, so I'll try to fix this up a bit while I'm in the area.

Reply:
If I recall correctly, the issue with modifying in place is that the file can be changed by other things while this UI is active but not updated, so then when you save you get problematic behavior. ntos has no concept of file locks.

Possible this isn't an issue anymore or can be worked around.

Reply:
I don't think this should be an issue, since the reports program uses a copy of the report it's editing, and the save function saves it locally. In the future some sort of file lock system is probably a good idea.

Reply:
Well, local as in on the local harddrive--but multiple users might still be accessing it at once from there.